### PR TITLE
Nessie: Bump Nessie and add Namespace support

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNamespace.java [hunk 1]
```diff
@@ -62,10 +72,35 @@ public void testListNamespaces() {
   }
 
   @Test
-  public void testDroppingNamespace() {
-    Assertions.assertThatThrownBy(() -> catalog.dropNamespace(Namespace.of("test")))
-        .isInstanceOf(UnsupportedOperationException.class)
-        .hasMessage("Cannot drop namespace 'test': dropNamespace is not supported by the NessieCatalog");
+  public void testCreatingAndDroppingNamespace() {
```
Comment (nastra) [inline nessie/src/test/java/org/apache/iceberg/nessie/TestNamespace.java:75]: additional Namespace testing will be added once we introduce a NessieCatalogTest

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java [hunk 2]
```diff
@@ -281,46 +287,74 @@ public void renameTable(TableIdentifier from, TableIdentifier toOriginal) {
...
 
-  /**
-   * creating namespaces in nessie is implicit, therefore this is a no-op. Metadata is ignored.
-   *
-   * @param namespace a multi-part namespace
-   * @param metadata  a string Map of properties for the given namespace
-   */
   @Override
   public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    try {
```
Comment (snazy) [inline nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:292]: Here (and dropNamespace below)  a
```java
    reference.checkMutable();
```
is missing. And after the creation/deletion of the namespace, `reference` must be updated to point to the new HEAD.

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java [hunk 3]
```diff
@@ -281,46 +287,74 @@ public void renameTable(TableIdentifier from, TableIdentifier toOriginal) {
...
+      api.createNamespace()
+          .refName(reference.getName())
```
Comment (snazy) [inline nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:294]: Should be `.reference(reference)` (for all other occurences below as well)

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java [hunk 4]
```diff
@@ -281,46 +287,74 @@ public void renameTable(TableIdentifier from, TableIdentifier toOriginal) {
...
+          .namespace(org.projectnessie.model.Namespace.of(namespace.levels()))
+          .create();
+    } catch (NessieNamespaceAlreadyExistsException e) {
+      throw new AlreadyExistsException(e, "Namespace '%s' already exists.", namespace);
+    } catch (NessieReferenceNotFoundException e) {
+      LOG.error("Cannot create Namespace '{}': ref is no longer valid.", namespace, e);
```
Comment (snazy) [inline nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:300]: Should actually be a RuntimeException and not a silent catch (even with a log)

**Review by snazy — CHANGES_REQUESTED** [review]
Namespace operations must use the Nessie `reference` including the hash. It doesn't matter much for now, but once namespaces have properties, it probably matters.

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java (hunk 4 above)
Comment (nastra) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:300]: done

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java (hunk 3 above)
Comment (nastra) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:294]: done

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java (hunk 2 above)
Comment (nastra) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:292]: done

**Review by snazy — APPROVED** [review]

File: nessie/src/test/java/org/apache/iceberg/nessie/TestNamespace.java (hunk 1 above)
Comment (nastra) [reply nessie/src/test/java/org/apache/iceberg/nessie/TestNamespace.java:74]: https://github.com/apache/iceberg/pull/4392

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java [hunk 5]
```diff
@@ -281,46 +287,78 @@ public void renameTable(TableIdentifier from, TableIdentifier toOriginal) {
...
 
-  /**
-   * creating namespaces in nessie is implicit, therefore this is a no-op. Metadata is ignored.
-   *
-   * @param namespace a multi-part namespace
-   * @param metadata  a string Map of properties for the given namespace
-   */
   @Override
   public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    try {
```
Comment (rymurr) [inline nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:292]: I notice metadata is ignored but the javadoc is removed, can you add it back.

Comment (nastra) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:292]: I have a follow-up PR for that part of the code and will add that note there if that's ok?

**Review by rymurr — APPROVED** [review]

**Pull request merged**